### ISSUE TITLE
Feature: Warn on overreaching error suppression

### ISSUE THREAD

**Issue by cburgmer** [koalaman/shellcheck#1111]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
set -eo pipefail

grep -ve '^$' < "$FILE" || true
```

#### Here's what shellcheck currently says:
```
$ shellcheck myscript
No issues detected!
```

#### Here's what I wanted or expected to see:

This example script will silently fail if `$FILE` does not exist, due to `|| true` hiding the error. The goal of this script is to ignore grep failing, and the initial code was `echo "$FILE" | grep -ve '^$' || true` which shellcheck complained about (SC2002).

The proper solution IMHO would be
```sh
#!/bin/bash
set -eo pipefail

(grep -ve '^$'  || true) < "$FILE"
```

Shellcheck should fail with an error a long the lines of "overreaching error suppression".

If people agree on the issue and on the proposed solution, I would be happy to try a PR.
